### ISSUE TITLE
Git Resource Failure after upgrading to 5.1

### ISSUE THREAD

**Issue by ameowlia** [concourse/concourse#3843]
## Bug Report

Hello Concourse Folks! We are with the networking program at cloudfoundry. We recently upgraded to 5.1 from 4.2. After upgrading, two of our git resources (in multiple pipelines) started failing. The resources had previously succeeded with the same sha in previous runs before upgrading. It is failing on something we are vendoring in. This is the error we are seeing:
```
error: pathspec 'src/gopkg.in/gorp.v1' did not match any file(s) known to git.
```

We tried getting onto the container, and we were able to successfully run a recursive git submodule sync. 

Have you seen this issue before? Any tips for debugging the git resource? Is there a way to change the git resource version?

## Steps to Reproduce

1. Create this pipeline on course with version 5.1: 

```
resources:
  - name: cf-networking-oss
     type: git
     source:
          branch: master
          uri: https://github.com/cloudfoundry/cf-networking-release.git

jobs:
 - name: test-git-resource
    plan:
    - get: cf-networking-oss
       trigger: true
```
2. See it fail.

3. Create this pipeline on version with 4.2
4. See it pass.

## Expected Results

The git resource is able to get the code without failing.

## Actual Results

The git resource fails when getting the code

## Additional Context

This is blocking our pipelines and preventing us from releasing. We thought about going back to an older version of concourse, but we are worried that there were probably data migrations and undoing that might get messy.

## Version Info

* Concourse version: 5.1.0
* Deployment type (BOSH/Docker/binary): BOSH
* Infrastructure/IaaS: GCP
* Did this used to work? Yes

## Tagging Fun
@KauzClay

or ping us on slack at #networking in the cloud foundry slack.

**Comment by Typositoire** [comment]
Actually I was able to make it pass using the example pipeline you provided

![image](https://user-images.githubusercontent.com/721017/57480302-52f06a00-726d-11e9-892d-ef4ea3173925.png)

**Comment by Typositoire** [comment]
I'd be curious to see the logs of your runs.

**Comment by vito** [comment]
Sounds like the same thing as https://github.com/concourse/git-resource/issues/259 - this is fixed in Git resource v1.4 which will be in Concourse v5.2 (shipping soon).
